packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -440,10 +452,13 @@ public void run(Future<String> tokenFuture) {
                 result.success(tokenResult);
               } catch (ExecutionException e) {
                 if (e.getCause() instanceof UserRecoverableAuthException) {
-                  if (shouldRecoverAuth) {
-                    // Move this to top of getTokens method.
+                  if (shouldRecoverAuth && pendingOperation == null) {
                     checkAndSetPendingOperation(METHOD_GET_TOKENS, result, email);
-                    recoverAuthFromException((UserRecoverableAuthException) e.getCause());
+                    Intent recoveryIntent =
```

Comment:
I did a bit of a cleanup too. This piece of code actually executes on the UiThread so there's no need to wrap it in another UiThread executor.

Reply:
Good catch. I hadn't realized that until I read the docs for `BackgroundTaskRunner`.